dagger/pipeline/tasks/reverse_etl_task.py

```diff
@@ -0,0 +1,246 @@
+        )
+
+    def __init__(self, name, pipeline_name, pipeline, job_config):
+        super().__init__(name, pipeline_name, pipeline, job_config)
+
+        self._executable = self.executable or self.DEFAULT_EXECUTABLE
+        self._executable_prefix = self.executable_prefix or self.DEFAULT_EXECUTABLE_PREFIX
+
+        self._assume_role_arn = self.parse_attribute("assume_role_arn")
+        self._num_threads = self.parse_attribute("num_threads") or self.DEFAULT_NUM_THREADS
+        self._batch_size = self.parse_attribute("batch_size") or self.DEFAULT_BATCH_SIZE
+        self._absolute_job_name = self._absolute_job_name or self.DEFAULT_JOB_NAME
+        self._primary_id_column = self.parse_attribute("primary_id_column")
+        self._secondary_id_column = self.parse_attribute("secondary_id_column")
+        self._custom_id_column = self.parse_attribute("custom_id_column")
+        self._model_name = self.parse_attribute("model_name")
+        self._project_name = self.parse_attribute("project_name") or self.DEFAULT_PROJECT_NAME
+        self._is_deleted_column = self.parse_attribute("is_deleted_column")
+        self._hash_column = self.parse_attribute("hash_column")
+        self._updated_at_column = self.parse_attribute("updated_at_column")
+        self._from_time = self.parse_attribute("from_time")
+        self._days_to_live = self.parse_attribute("days_to_live")
+
+        if self._hash_column and self._updated_at_column:
+            raise ValueError(f"ReverseETLTask: {self._name} hash_column and updated_at_column are mutually exclusive")
+
+        if self._hash_column or self._updated_at_column:
+            if not self._from_time:
+                raise ValueError(f"ReverseETLTask: {self._name} from_time is required when hash_column or updated_at_column is provided")
+
+        # Making sure the input table name is set as it is expected in the reverse etl job
+        input_index = self._get_io_index(self._inputs)
+        print('XXX', self._inputs, input_index)
+        if input_index is None:
+            raise ValueError(f"ReverseEtlTask: {self._name} must have an input")
+        self._inputs[input_index].name = "input_table_name"
+
+        # Making sure the output name is set as it is expected in the reverse etl job
+        output_index = self._get_io_index(self._outputs)
+        if output_index is None:
+            raise ValueError(f"ReverseEtlTask: {self._name} must have an output")
+        self._outputs[output_index].name = "output_name"
+
+        # Extracting the output type from the output definition
+        self._output_type = self._outputs[output_index].ref_name
+
+        # Extracting the outputs region name from the output definition
+        self._region_name = self._outputs[output_index].region_name
+
+
+    def _get_io_index(self, ios):
+        if len([io for io in ios if io.ref_name != "dummy"]) > 1:
+            raise ValueError(f"ReverseEtlTask: {self._name} can only have one input or output")
```

Comment:
We dont want to support the case where there are multiple inputs?
(for example a simple union of two tables)

Reply:
We want, but atm we don't